apiserver/metricsmanager/metricsmanager.go

```diff
@@ -143,25 +146,42 @@ func (api *MetricsManagerAPI) AddJujuMachineMetrics() error {
 		return errors.Trace(err)
 	}
 	machineCount := 0
+	osMachineCount := map[os.OSType]int{}
 	for _, machine := range allMachines {
 		ct := machine.ContainerType()
 		if ct == instance.NONE || ct == "" {
 			machineCount++
+			if osType, err := series.GetOSFromSeries(machine.Series()); err != nil {
+				logger.Warningf("failed to resolve OS name for series %q: %v", machine.Series(), err)
```

Comment:
IMO we should count these as "unknown OS".
We can still show the tally to the user. Otherwise the numbers won't add up.